src/runtime_src/core/common/system.cpp

```diff
@@ -176,4 +176,10 @@ get_monitor_access_type()
   return instance().get_monitor_access_type();
 }
 
+void
+program_plp(std::shared_ptr<device> dev, std::vector<char> buffer)
```

Comment:
Why is this a system function?  It looks like a device function?

Why are the arguments passed by value?  That is a lot of copying for the buffer.